contracts/cw1-subkeys/src/contract.rs

```diff
@@ -1187,14 +1189,14 @@ mod tests {
         let res = handle(&mut deps, env.clone(), handle_msg.clone()).unwrap();
         assert_eq!(res.messages, msgs);
         assert_eq!(
-            res.log,
-            vec![log("action", "execute"), log("owner", spender1.clone())]
+            res.attributes,
+            vec![attr("action", "execute"), attr("owner", spender1.clone())]
         );
 
         // And then cannot (not enough funds anymore)
         let res = handle(&mut deps, env, handle_msg.clone());
         match res.unwrap_err() {
-            StdError::Underflow { .. } => {}
+            ContractError::Std(StdError::Underflow { .. }) => {}
```

Comment:
But since we are are creating custom errors everywhere, maybe we want to map this.
Either where it is generated like `.map_err(|e| ...)`. Or do something like:

```rust
impl From<cw1_whitelist::ContractError> for ContractError
    fn from(err: cw1_whitelist::ContractError) -> Self {
        match err {
            cw1_whitelist::ContractError::Std(error) => ContractError::Std(error),
            cw1_whitelist::ContractError::Unauthorized {} => ContractError::Unauthorized {},
        }
    }
}
```

But mapping `StdError::Underflow{}` => `ContractError::InsufficientAllowance{}`?

Reply:
Although the underflow message does show you the amounts, which is nice and shouldn't be lost.

This is an idea, take it or leave it.

Reply:
I would leave it as it is. Besides the extra boiler plate, it's pretty clear what's being done.